src/test/scala/mesosphere/marathon/raml/VolumeConversionTest.scala

```diff
@@ -109,61 +109,67 @@ class VolumeConversionTest extends UnitTest {
     }
   }
 
-  "RAML persistent volume conversion" when {
-    val volume = AppPersistentVolume(
-      "/container",
-      PersistentVolumeInfo(None, size = 123L, maxSize = Some(1234L), profileName = Some("ssd-fast"),
-        constraints = Set.empty),
-      ReadMode.Rw)
-    "converting from RAML" should {
-      val (persistent, mount) = Some(volume.fromRaml).collect {
-        case state.VolumeWithMount(v: state.PersistentVolume, m) => (v, m)
-      }.getOrElse(fail("expected PersistentVolume"))
-      "convert all fields to core" in {
-        mount.mountPath should be(volume.containerPath)
-        mount.readOnly should be(false)
-        persistent.persistent.`type` should be(state.DiskType.Root)
-        persistent.persistent.size should be(volume.persistent.size)
-        persistent.persistent.maxSize should be(volume.persistent.maxSize)
-        persistent.persistent.profileName should be(volume.persistent.profileName)
-        persistent.persistent.constraints should be(Set.empty)
+  Seq(Some("ssd-fast"), None).foreach { profileName =>
```

Comment:
I would prefer up if we did:

```suggestion
  Seq(Some("ssd-fast") -> state.DiskType.Mount, None -> state.DiskType.Root).foreach { case (profileName, expectedType) =>
```

Otherwise, we have code in the test that looks like the implementation.

Reply:
Yeah I had the same feeling here …